app/catalog_data.go

```diff
@@ -16,23 +23,147 @@ const (
 )
 
-func addCatalogs(management *config.ManagementContext) error {
+// updateCatalogURL checks the default system-library version against the catalog actual version
+// and annotated version, then updates catalog's version and its annotation if an outdated version
+// is used.
+func updateCatalogURL2(catalogClient v3.CatalogInterface, desiredDefaultURL string, desiredDefaultBranch string) error {
+	update := false
+
+	catalog, err := catalogClient.Get(systemLibraryName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Update if there is no annotation and url/branch are old defaults
+	if anno := catalog.Annotations[defSystemChartVer]; anno == "" {
+		update = true
+		if catalog.Spec.URL == systemLibraryURL && catalog.Spec.Branch == systemLibraryBranch {
+			if catalog.Spec.URL != desiredDefaultURL || catalog.Spec.Branch != desiredDefaultBranch {
+				setDesiredChartLib(catalog, desiredDefaultURL, desiredDefaultBranch)
+			}
+		}
+	} else {
+		annoMap := make(map[string]interface{})
+		err := json.Unmarshal([]byte(anno), &annoMap)
+		if err != nil {
+			update = true
+			logrus.Errorf("%s annotation should be in JSON format: %v", defSystemChartVer, err)
+		}
+
+		if annoMap["url"] != desiredDefaultURL || annoMap["branch"] != desiredDefaultBranch {
+			update = true
+		}
+
+		// Update if annotation matches catalog spec, but neither match environment branch
+		if catalog.Spec.URL != desiredDefaultURL || catalog.Spec.Branch != desiredDefaultBranch {
+			if catalog.Spec.URL == annoMap["url"] && catalog.Spec.Branch == annoMap["branch"] {
+				setDesiredChartLib(catalog, desiredDefaultURL, desiredDefaultBranch)
+			}
+		}
+	}
+
+	if catalog.Annotations == nil {
+		catalog.Annotations = make(map[string]string)
+	}
+
+	if update {
+		setCatalogAnnotation(catalog, desiredDefaultURL, desiredDefaultBranch)
+	}
+	catalogClient.Update(catalog)
+
+	return nil
+}
+
+// updateCatalogURL updates annotations if they are outdated and system catalog url/branch if it matches outdated defaults
+func updateCatalogURL(catalogClient v3.CatalogInterface, desiredDefaultURL string, desiredDefaultBranch string) error {
+	oldCatalog, err := catalogClient.Get(systemLibraryName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	desiredCatalog := oldCatalog.DeepCopy()
+
+	if oldAnno := oldCatalog.Annotations[defSystemChartVer]; oldAnno == "" {
+		// If url/branch are old defaults, update - otherwise they are user set and should not be changed
+		if oldCatalog.Spec.URL == systemLibraryURL && oldCatalog.Spec.Branch == systemLibraryBranch {
+			setDesiredChartLib(desiredCatalog, desiredDefaultURL, desiredDefaultBranch)
+		}
+	} else {
+		oldAnnotations := make(map[string]interface{})
+		json.Unmarshal([]byte(oldAnno), &oldAnnotations)
+		if err != nil {
+			logrus.Errorf("%s annotation should be in JSON format: %v", defSystemChartVer, err)
+		}
+
+		// If url/branch catalog spec does not match environment but matches old annotation, update to new values
```

Comment:
Update this comment `// if spec and anno match, user hasn't touched it hopefully so use the new env` < something like this